src/openeo_aggregator/backend.py

```diff
@@ -125,53 +152,91 @@ def _merge_collection_metadata(self, by_backend: Dict[str, dict]) -> dict:
-        keywords = getter.union("keywords", skip_duplicates=True)
-        if keywords:
-            result["keywords"] = keywords
+        # Note: CRS is required by OGC API: https://docs.opengeospatial.org/is/18-058/18-058.html#_crs_identifier_list
+        result.update(getter.simple_merge([
+            "stac_extensions", "keywords", "deprecated", "providers", "assets",
+            "crs",
+            "sci:citation", "sci:doi", "sci:publications"
+        ]))
+
+        ## All keys with special merge handling.
         versions = set(getter.get("version"))
         if versions:
-            # TODO: smarter version maximum?
+            # TODO: smarter version maximum? Low priority, versions key is not used in most backends.
             result["version"] = max(versions)
-        deprecateds = list(getter.get("deprecated"))
-        if deprecateds:
-            result["deprecated"] = all(deprecateds)
         licenses = set(getter.get("license"))
         result["license"] = licenses.pop() if len(licenses) == 1 else ("various" if licenses else "proprietary")
-        # TODO: .. "various" if multiple licenses apply ... links to the license texts SHOULD be added,
-        providers = getter.union("providers", skip_duplicates=True)
-        if providers:
-            result["providers"] = list(providers)
+
         result["extent"] = {
             "spatial": {
-                "bbox": getter.select("extent").select("spatial").union("bbox", skip_duplicates=True) \
+                "bbox": getter.select("extent").select("spatial").merge_arrays("bbox", skip_duplicates=True) \
                         or [[-180, -90, 180, 90]],
             },
             "temporal": {
-                "interval": getter.select("extent").select("temporal").union("interval", skip_duplicates=True) \
+                "interval": getter.select("extent").select("temporal").merge_arrays("interval", skip_duplicates=True) \
                             or [[None, None]],
             },
         }
-        result["links"] = list(getter.union("links"))
-        # TODO: combine cube:dimensions smarter?
+
         cube_dimensions = getter.first("cube:dimensions")
         if cube_dimensions:
             result["cube:dimensions"] = cube_dimensions
-        # TODO merge existing summaries?
-        result["summaries"] = {
-            # TODO: use a more robust/user friendly backend pointer than backend id (which is internal implementation detail)
-            self.STAC_PROPERTY_PROVIDER_BACKEND: list(by_backend.keys())
-        }
-        # TODO: assets ?
-
+            cube_band_value = getter.select("cube:dimensions").first("bands", None)
+            # Bands
+            if cube_band_value is not None:
+                result["cube:dimensions"]["bands"] = cube_band_value
+                cube_dimension_bands = list(getter.select("cube:dimensions").select("bands").merge_arrays('values', skip_duplicates=True))
```

Comment:
just concatenating band values here will give unexpected results (as band indexes possibly change, confusing client side handling)

Reply:
handled: now just common prefix is used